pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -322,14 +315,23 @@ func (e *cloudWatchExecutor) newSession(ctx context.Context, pluginCtx backend.P
 			}
 			sess.Config.HTTPClient.Transport = defTransport.Clone()
 		}
-		err = proxy.New(instance.ProxyOpts).ConfigureSecureSocksHTTPProxy(sess.Config.HTTPClient.Transport.(*http.Transport))
+		err = proxy.New(ds.ProxyOpts).ConfigureSecureSocksHTTPProxy(sess.Config.HTTPClient.Transport.(*http.Transport))
 		if err != nil {
 			return nil, fmt.Errorf("error configuring Secure Socks proxy for Transport: %w", err)
 		}
 	}
 	return sess, nil
 }
 
+func (e *cloudWatchExecutor) newSessionFromContext(ctx context.Context, pluginCtx backend.PluginContext, region string) (*session.Session, error) {
```

Comment:
Why would it be contexts? Sorry I don't think I follow!

Reply:
I was thinking Contexts because it gets the instance from `ctx` and `pluginCtx` but this isn't an exported function so the naming doesn't matter that much as long as we get it.

Reply:
ah I see, I guess I have no preference!